### PR TITLE
ScriptManager migration to the CodeAnalysis class. This is in preparation for the net 6 migration.

### PR DESCRIPTION
**Pull request by pzgulyas**
I have accidentally closed the PR #553 and could not reopen it. So I created this one as a replacement.
Blueprint: https://blueprints.launchpad.net/or/+spec/dependencies-update
("Switch to new C# script compiler implementation")

### CODE REVIEW

**Comment by pzgulyas** [comment]
Cédric @Sharpe49, thanks for the modification with the debug symbols, I committed your changes here.

File: Source/Orts.Simulation/packages.config [hunk 1]
```diff
@@ -1,4 +1,16 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
+  <package id="Microsoft.CodeAnalysis.Analyzers" version="3.3.2" targetFramework="net472" developmentDependency="true" />
+  <package id="Microsoft.CodeAnalysis.Common" version="4.0.1" targetFramework="net472" />
+  <package id="Microsoft.CodeAnalysis.CSharp" version="4.0.1" targetFramework="net472" />
   <package id="Microsoft.CodeDom.Providers.DotNetCompilerPlatform" version="3.6.0" targetFramework="net472" />
```
Comment (Sharpe49) [inline Source/Orts.Simulation/packages.config:6]: Please remove the old package

File: Source/Orts.Simulation/Orts.Simulation.csproj [hunk 2]
```diff
@@ -198,14 +230,21 @@
...
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
   <Import Project="..\packages\Microsoft.CodeDom.Providers.DotNetCompilerPlatform.3.6.0\build\net472\Microsoft.CodeDom.Providers.DotNetCompilerPlatform.targets" Condition="Exists('..\packages\Microsoft.CodeDom.Providers.DotNetCompilerPlatform.3.6.0\build\net472\Microsoft.CodeDom.Providers.DotNetCompilerPlatform.targets')" />
   <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
     <PropertyGroup>
       <ErrorText>Ce projet fait référence à des packages NuGet qui sont manquants sur cet ordinateur. Utilisez l'option de restauration des packages NuGet pour les télécharger. Pour plus d'informations, consultez http://go.microsoft.com/fwlink/?LinkID=322105. Le fichier manquant est : {0}.</ErrorText>
     </PropertyGroup>
     <Error Condition="!Exists('..\packages\Microsoft.CodeDom.Providers.DotNetCompilerPlatform.3.6.0\build\net472\Microsoft.CodeDom.Providers.DotNetCompilerPlatform.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.CodeDom.Providers.DotNetCompilerPlatform.3.6.0\build\net472\Microsoft.CodeDom.Providers.DotNetCompilerPlatform.targets'))" />
```
Comment (Sharpe49) [inline Source/Orts.Simulation/Orts.Simulation.csproj:243]: Please remove the old package.

**Review by Sharpe49 — DISMISSED** [review]

File: Source/Orts.Simulation/packages.config (hunk 1 above)
Comment (pzgulyas) [reply Source/Orts.Simulation/packages.config:6]: Ah, you are absolutely right. I'm travelling on this weekend though, so probably won't be able to do it for some time, sorry for that. (Feel  free to push a commit, if this is the barrier for merging.)

**Comment by cjakeman** [comment]
Hi Peter, 

Which Blueprint does this PR refer to please?

Thakns,

Chris

**Comment by pzgulyas** [comment]
> Which Blueprint does this PR refer to please?
The blueprint is: https://blueprints.launchpad.net/or/+spec/dependencies-update
point 2: "Switch to new C# script compiler implementation"

**Comment by cesarBLG** [comment]
There's yet another reference to old compiler at https://github.com/pzgulyas/openrails-1/blob/ec2fd8581a22e95eec855505a170911510669685/Source/Orts.Simulation/Orts.Simulation.csproj#L56-L58

**Comment by pzgulyas** [comment]
> There's yet another reference to old compiler at https://github.com/pzgulyas/openrails-1/blob/ec2fd8581a22e95eec855505a170911510669685/Source/Orts.Simulation/Orts.Simulation.csproj#L56-L58

Thanks, I removed that. Hope no more left.

**Review by cesarBLG — APPROVED** [review]
Thanks Peter, it seems OK.

As an extra, with this new implementation scripts now run well on Linux under Wine.

**Comment by pzgulyas** [comment]
Ah, glad to hear this positive side-effect, thanks!

**Pull request merged**